semgrep-core/src/engine/Match_search_rules.ml

```diff
@@ -989,15 +989,5 @@ let check ~match_hook default_config rules file_and_more =
   |> List.map (fun (r, pformula) ->
          let rule_id = fst r.R.id in
          Common.profile_code (spf "real_rule:%s" rule_id) (fun () ->
-             try
-               check_rule r match_hook default_config pformula file_and_more
-               (* TODO: why do we intercept errors here? We already do that in
-                * Run_semgrep.ml. *)
-             with exn when not !Flag_semgrep.fail_fast ->
-               {
-                 RP.matches = [];
-                 errors = [ E.exn_to_error ~rule_id:(Some rule_id) file exn ];
```

Comment:
Hm I think it might have been to attach the rule_id, we'll probably have some failing tests

Reply:
We can do the same trick I did in Run_semgrep.ml where there's a global that remember the last mini_ruleid and set it in Match_search_rules.ml and use it in Run_semgrep.ml